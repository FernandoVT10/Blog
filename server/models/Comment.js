import mongoose, { Schema } from "mongoose";

const comment = new Schema({
    articleId: {
        type: mongoose.Types.ObjectId,
        ref: "articles"
    },
    name: {
        type: String,
        maxlength: 30,
        required: true
    },
    comment: {
        type: String,
        maxlength: 500,
        required: true
    }
}, { timestamps: true });

export default mongoose.model("comments", comment)