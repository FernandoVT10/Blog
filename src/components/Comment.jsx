import "../styles/components/comment.scss";

export default ({ comment }) => {
    return (
        <div className="comment">
            <p className="comment__name">
                { comment.name }
            </p>
            <p className="comment__content">
                { comment.comment }
            </p>
        </div>
    );
};