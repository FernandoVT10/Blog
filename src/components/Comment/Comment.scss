@use "../../styles/variables";

.comment {
    margin-top: 25px;
    width: 100%;
    padding: 20px;
    background: variables.$black;

    &__name {
        color: white;
        font-size: 18px;
        font-weight: bold;
    }

    &__content {
        color: white;
        margin-bottom: 0;
    }
}