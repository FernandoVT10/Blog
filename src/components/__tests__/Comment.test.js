import { create } from "react-test-renderer";
import Comment from "../Comment/Comment";

const COMMENT_MOCK = {
    name: "Test name",
    comment: "This is a test"
};

test("Check if <Comments/> renders correctly", () => {
    const component = create(<Comment comment={COMMENT_MOCK}/>);

    expect(component.toJSON()).toMatchSnapshot();
});