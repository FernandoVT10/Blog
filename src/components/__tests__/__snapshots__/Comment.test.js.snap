// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if <Comments/> renders correctly 1`] = `
<div
  className="comment"
>
  <p
    className="comment__name"
  >
    Test name
  </p>
  <p
    className="comment__content"
  />
</div>
`;
